### ISSUE TITLE
知识库documents post服务阻塞其余api访问

### ISSUE THREAD

**Issue by 401557122** [langgenius/dify#5540]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.6.8

### Cloud or Self Hosted

Self Hosted (Source)

### Steps to reproduce

您好，我是使用docker-compose.middleware.yaml部署的相关中间件，并且使用源码运行flask run --host 0.0.0.0 --port=5001以及celery -A app.celery worker -P gevent -c 1 -Q dataset,generation,mail --loglevel INFO，并使用web的镜像直接搭建前端。
在使用中我发现，当我点击页面的知识库中的保存并处理按钮，也就是documents接口或者datasets/init接口，因为文档多或文档内容多的时候，会持续处理，此时，其余的智能体、页面相关接口，调用均是卡死的状态无法返回内容。
请问这应该如何解决，我起初以为是celery worker设置的1，我改为128仍出现该问题

### ✔️ Expected Behavior

希望文档处理任务在运行时互不影响，且不影响其余服务调用

**Comment by 401557122** [comment]
你太帅了机器哥

**Comment by SeesawLiu** [comment]
I configured it according to the above method, but it still doesn't work. I configured unstructured, and after uploading the file, 1. First, api cpu 100%, unstructured cpu usage is low, worker_general cpu usage is low, and the front end cannot be used. 2. Then worker_general execution 100%, worker_api cpu usage is low, and the front end can be used. 3. worker_general cpu 50% worker_api cpu 100%

@dosu
